evaluation/benchmarks/gaia/run_infer.py

```diff
@@ -70,18 +73,34 @@ def get_config(
         workspace_base=None,
         workspace_mount_path=None,
     )
-    config.set_llm_config(metadata.llm_config)
+    config.set_llm_config(
+        update_llm_config_for_completions_logging(
+            metadata.llm_config, metadata.eval_output_dir, instance['instance_id']
+        )
+    )
+
+    config_copy = copy.deepcopy(config)
+    load_from_toml(config_copy)
+    model_routing_config = config_copy.get_agent_config().model_routing
+    # Set log_completions to True for all routing LLMs
+    for llm_cfg in model_routing_config.routing_llms.values():
+        llm_cfg.log_completions = True
+        update_llm_config_for_completions_logging(
+            llm_cfg, metadata.eval_output_dir, instance['instance_id']
+        )
```

Comment:
This seems a lot of code just to turn on routing -- is config_copy necessary? can we simplify?